server/src/main/java/org/apache/druid/server/QueryResource.java

```diff
@@ -156,13 +156,19 @@ public Response cancelQuery(@PathParam("id") String queryId, @Context final Http
   public Response doPost(
       final InputStream in,
       @QueryParam("pretty") final String pretty,
-      @Context final HttpServletRequest req // used to get request content-type, remote address and auth-related headers
+      @Context final HttpServletRequest req // used to get request content-type,Accept header, remote address and auth-related headers
   ) throws IOException
   {
     final QueryLifecycle queryLifecycle = queryLifecycleFactory.factorize();
     Query<?> query = null;
 
-    final ResponseContext context = createContext(req.getContentType(), pretty != null);
+    String acceptHeader = req.getHeader("Accept");
+    if (acceptHeader == null || acceptHeader.length() == 0) {
```

Comment:
(minor) `Strings.isNullOrEmpty` is usually used in such a scenario

Reply:
hi @drcrallen - i changed to Strings.isNullOrEmpty. commit pushed.